lib/shopify_api/only_sync_dirty_attributes.rb

```diff
@@ -0,0 +1,34 @@
+module ShopifyAPI
+  module OnlySyncDirtyAttributes
+    def method_missing(method_symbol, *args)
+      method_name = method_symbol.to_s
+
+      if match_data = /(?<attribute_name>.+)(?:=)$/.match(method_name)
+        changed << match_data[:attribute_name]
+      end
+
+      super
+    end
+
+    def save
+      slice_attributes_on_changed
```

Comment:
I would block this

```rb
def save(*)
  with_dirty_attributes { super }
end

def with_dirty_attributes
  @old_attributes = @attributes
  @attributes.slice(...)
rescue # ???
  # could potentially restore the state in case something goes awfully wrong, nothing I can't think of right now, but it would be fairly easy to add.
ensure
  @changed.clear
end
```